external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -304,6 +304,14 @@ pub struct ParsingSymbolicator<'a> {
     file_id_mapping: &'a HashMap<FileHash, usize>,
     // A mapping from file IDs to a split vector of the lines in each file (used to build docstrings)
     file_id_to_lines: &'a HashMap<usize, Vec<String>>,
+    /// Associates uses for a given definition to allow displaying all references
+    references: &'a mut BTreeMap<DefLoc, BTreeSet<UseLoc>>,
+    /// A UseDefMpa for a given module (needs to be appropriately set before the module
+    /// processing starts)
+    use_defs: UseDefMap,
```

Comment:
why not having this tracked on a side struct instead of a `ModuleDefs` struct?
I mean something like def-use/use-def object.

Reply:
I am not sure I understand. The `use_defs` map is updated both in the `ParsingSymbolicator` and in `TypingSymbolicator` so it's part of both now so that it's easy to access via `self` wherever in the AST processing code we are. That being said, there is no super deep thinking in how the symbolicators are structured, particularly that they evolved over time and new pieced of meta data that we needed to track/update where added as needed.